test/test_rwlock.ml

```diff
@@ -0,0 +1,81 @@
+open QCheck
+open STM
+module Rwlock = Picos_std_sync.Rwlock
+
+module Spec = struct
+  include SpecDefaults
+  include Wrap
+
+  type cmd = Push of int | Pop_opt | Top_opt | Length
+
+  let show_cmd = function
+    | Push x -> "Push " ^ string_of_int x
+    | Pop_opt -> "Pop_opt"
+    | Top_opt -> "Top_opt"
+    | Length -> "Length"
+
+  module State = struct
+    type t = int list
+
+    let push x xs = x :: xs
+    let pop_opt = function _ :: xs -> xs | [] -> []
+    let top_opt = function x :: _ -> Some x | [] -> None
+    let length = List.length
+  end
+
+  type state = State.t
+  type sut = { stack : int Stack.t; rwlock : Rwlock.t }
+
+  let arb_cmd _s =
+    [
+      Gen.int_range 1 1000 |> Gen.map (fun x -> Push x);
+      Gen.return Pop_opt;
+      Gen.return Top_opt;
+      Gen.return Length;
+    ]
+    |> Gen.oneof |> make ~print:show_cmd
+
+  let init_state = []
+  let init_sut () = { stack = Stack.create (); rwlock = Rwlock.create () }
+  let cleanup _ = ()
```

Comment:
Drive-by-comment, noticed while studying this use case of https://github.com/ocaml-multicore/multicoretests/pull/511:
```suggestion
```

Reply:
Thanks!